pkg/connector/org_role.go

```diff
@@ -0,0 +1,400 @@
+}
+
+func (o *orgRoleResourceType) List(
+	ctx context.Context,
+	parentID *v2.ResourceId,
+	pToken *pagination.Token,
+) ([]*v2.Resource, string, annotations.Annotations, error) {
+	if parentID == nil {
+		return nil, "", nil, nil
+	}
+
+	bag, _, err := parsePageToken(pToken.Token, &v2.ResourceId{ResourceType: resourceTypeOrgRole.Id})
+	if err != nil {
+		return nil, "", nil, err
+	}
+
+	orgName, err := o.orgCache.GetOrgName(ctx, parentID)
+	if err != nil {
+		return nil, "", nil, err
+	}
+
+	roles, resp, err := o.client.Organizations.ListRoles(ctx, orgName)
+	if err != nil {
+		// Handle permission errors gracefully
+		if resp != nil && (resp.StatusCode == http.StatusForbidden || resp.StatusCode == http.StatusNotFound) {
+			// Return empty list with no error to indicate we skipped this resource
+			pageToken, err := bag.NextToken("")
+			if err != nil {
+				return nil, "", nil, err
+			}
+			return nil, pageToken, nil, nil
+		}
+		return nil, "", nil, fmt.Errorf("failed to list organization roles: %w", err)
+	}
+
+	var ret []*v2.Resource
+	for _, role := range roles.CustomRepoRoles {
+		roleResource, err := orgRoleResource(ctx, &OrganizationRole{
+			ID:          role.GetID(),
+			Name:        role.GetName(),
+			Description: role.GetDescription(),
+		}, &v2.Resource{Id: parentID})
+		if err != nil {
+			return nil, "", nil, err
+		}
+		ret = append(ret, roleResource)
+	}
+
+	pageToken, err := bag.NextToken("")
+	if err != nil {
+		return nil, "", nil, err
+	}
+
+	return ret, pageToken, nil, nil
+}
+
+func (o *orgRoleResourceType) Entitlements(
+	_ context.Context,
+	resource *v2.Resource,
+	_ *pagination.Token,
+) ([]*v2.Entitlement, string, annotations.Annotations, error) {
+	rv := make([]*v2.Entitlement, 0, 1)
+	rv = append(rv, entitlement.NewAssignmentEntitlement(resource, "assigned",
+		entitlement.WithDisplayName(resource.DisplayName),
+		entitlement.WithDescription(fmt.Sprintf("Assignment to %s role in GitHub", resource.DisplayName)),
+		entitlement.WithAnnotation(&v2.V1Identifier{
+			Id: fmt.Sprintf("org_role:%s", resource.Id.Resource),
+		}),
+		entitlement.WithGrantableTo(resourceTypeUser),
+	))
+
+	return rv, "", nil, nil
+}
+
+func (o *orgRoleResourceType) Grants(
+	ctx context.Context,
+	resource *v2.Resource,
+	pToken *pagination.Token,
+) ([]*v2.Grant, string, annotations.Annotations, error) {
+	if resource == nil {
+		return nil, "", nil, nil
+	}
+
+	bag, _, err := parsePageToken(pToken.Token, &v2.ResourceId{ResourceType: resourceTypeOrgRole.Id})
+	if err != nil {
+		return nil, "", nil, err
+	}
+
+	orgName, err := o.orgCache.GetOrgName(ctx, resource.ParentResourceId)
+	if err != nil {
+		return nil, "", nil, err
+	}
+
+	roleID, err := strconv.ParseInt(resource.Id.Resource, 10, 64)
+	if err != nil {
+		return nil, "", nil, fmt.Errorf("invalid role ID: %w", err)
+	}
+
+	var ret []*v2.Grant
+
+	// First, get teams with this role
+	teams, resp, err := o.client.Organizations.ListTeamsAssignedToOrgRole(ctx, orgName, roleID, nil)
```

Comment:
I think we will want to use a pagination bag so we can get team assignments/user assignments separately so we paginate properly. An example of how we do this: https://github.com/ConductorOne/baton-okta/blob/81354f306dc69ed74754499a8ea8bb5b234b6943/pkg/connector/group.go#L124

Reply:
I pushed a new commit to address the pagination